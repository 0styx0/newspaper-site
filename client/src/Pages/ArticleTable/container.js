import React from 'react';
import fetchFromApi from '../../helpers/fetchFromApi';

import ArticleTable from './';

export default class ArticleTableContainer extends React.Component {

    constructor() {
        super();

        this.putData = this.putData.bind(this);

        this.state = {
            history: []
        };
    }

    async putData(num = null) {

        const lastPath = +window.location.pathname.split("/").slice(-1);

        num = (isNaN(lastPath) || ((!isNaN(num)) && num)) ? num : lastPath;


        // use cached results (don't load new info)
        if (this.state.history[num]) {
            this.setState({
                articles: this.state.history[num].articles,
                issueInfo: this.state.history[num].issueInfo
            });

            window.history.pushState({}, `Issue ${num}`, num);
        }
        else {

            const data = await this.getData(num);
            const json = await data.json();

            window.history.pushState({}, `Issue ${num}`, isNaN(lastPath) ? "modifyArticles/"+json[2].num : json[2].num);

            this.setArticleInfo(json);
        }
    }

    componentWillMount() {

        this.putData(null);
    }

    async getData(num) {

         return await fetchFromApi(`articleGroup?articlesFor=${num}`);

    }


    setArticleInfo(data) {

        const history = [...this.state.history];
        history[data[2].num] = {articles: data[0], issueInfo: data[2]};

        this.setState({
            articles: data[0].length ? data[0] :  null,
            issueInfo: data[2],
            history
        });
    }

    render() {

            return <ArticleTable
                     articles={this.state.articles}
                     issue={this.state.issueInfo}
                     update={(e) => this.putData(e.target.value)}
                   />
    }
}
