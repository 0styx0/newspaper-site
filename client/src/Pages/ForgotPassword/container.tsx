import * as React from 'react';
import { graphql, withApollo } from 'react-apollo';
import { RecoverPassword } from '../../graphql/user';
import ForgotPassword from './';

interface Props {
    recoverPassword:
      (params: {
          query: typeof RecoverPassword,
          variables: { username: string, authCode: string, email: string }
        }
      ) => Promise<{message: string}>;
}

export class ForgotPasswordContainer extends React.Component<Props, {}> {

    constructor(props: Props) {
        super(props);

        this.onSubmit = this.onSubmit.bind(this);

    }

    /**
     * Sends data to server
     */
    onSubmit(target: HTMLFormElement) {

        this.props.recoverPassword({
            query: RecoverPassword,
            variables: {
                authCode: (target.querySelector('input[name=lastAuth]') as HTMLInputElement).value,
                username: (target.querySelector('input[name=username]') as HTMLInputElement).value,
                email: (target.querySelector('input[name=email]') as HTMLInputElement).value,
            }
        });
    }

    render() {
        return <ForgotPassword onSubmit={this.onSubmit} />;
    }
}

const ForgotPasswordContainerWithData =
    graphql(RecoverPassword, {name: 'recoverPassword'})(ForgotPasswordContainer as any);

export default withApollo(ForgotPasswordContainerWithData);
