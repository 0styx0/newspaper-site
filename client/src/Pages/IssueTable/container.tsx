import * as React from 'react';
import { compose, graphql } from 'react-apollo';
import { IssueQuery, IssueUpdate } from '../../graphql/issues';
import { Issue } from './interface.shared';
import IssueTable from './';

interface State {
    privateIssue: { // admins can change these (until public is true)
        public?: boolean;
        name?: string;
    };
    loaded: boolean;
}

interface Props {
    data: {
        loading: boolean;
        issues: Issue[]
    };
    mutate: Function;
}

/**
 * Creates a table of issues, in descending order with info of
 * number, name, datePublished, and views (sum of views of all articles in the issue)
 *
 * If user is admin (lvl 3) and issue is not public, user can change issue name and make it public.
 * In that case, @see IssueTable.allowEditsOfLastIssue for difference in UI
 */
export class IssueTableContainer extends React.Component<Props, State> {

    constructor() {
        super();

        this.onChangeIssueInfo = this.onChangeIssueInfo.bind(this);
        this.onSubmit = this.onSubmit.bind(this);

        this.state = {
            privateIssue: {},
            loaded: false
        };
    }

    /**
     * When graphql gets data from @see graphql/issues,
     * this turns it into 2d array of what will be table rows and saves it to state.issueInfo
     *
     * If current user is an admin, enables editing of most recent, unpublished issue @see this.allowEditsOfLastIssue
     */
    componentWillReceiveProps(props: Props) {

        if (!props.data.issues || this.props.data.issues.length > 0) {
            return;
        }

        this.setState({loaded: true});
    }

    /**
     * @param e {HTMLInputElement event}
     *
     * Adds input's value to this.state.privateIssue[input.name]
     *
     * @example if event.target = <input name="public" value="1" />, then after this, this.state.privateIssue.public = 1
     */
    onChangeIssueInfo(e: Event) {

        const target = e.target as HTMLInputElement;

        const privateIssue = Object.assign({}, this.state.privateIssue);
        privateIssue[target.name] = target.value;

        this.setState({
            privateIssue
        });
    }

    /**
     * @param e {HTMLFormElement event}
     *
     * Sends this.state.privateIssue data to server (so name and/or public status can be saved to db)
     */
    onSubmit(e: Event) {

        e.stopPropagation();
        e.preventDefault();

        this.props.mutate({
            variables: {
                name: this.state.privateIssue.name,
                public: !!this.state.privateIssue.public
            }
        });
    }

    render() {

        if (this.props.data.issues.length < 1) {
            return null;
        }

        return (
            <IssueTable
              key={+this.state.loaded}
              onSubmit={this.onSubmit}
              onChangeIssueInfo={this.onChangeIssueInfo}
              issues={this.props.data.issues}
            />
        );
    }
}


const IssueTableContainerWithData = compose(
    graphql(IssueQuery),
    graphql(IssueUpdate)
)(IssueTable as any);

export default IssueTableContainerWithData;