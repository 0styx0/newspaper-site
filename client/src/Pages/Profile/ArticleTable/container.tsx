import * as React from 'react';
import { graphql, withApollo, compose } from 'react-apollo';
import { ArticleDelete } from '../../../graphql/articles';
import { Article } from '../shared.interfaces';
import UserArticleTable from './';

interface Props {
    articles: Article[];
    deleteArticle: Function;
}

/**
 * Container for @see ./index.tsx
 */
export class UserArticleTableContainer extends React.Component<Props, {idsToDelete: Set<string>}> {

    constructor(props: Props) {
        super(props);

        this.onDelete = this.onDelete.bind(this);
        this.onSubmit = this.onSubmit.bind(this);

        this.state = {
            idsToDelete: new Set<string>()
        };
    }

    /**
     * Adds an article to state.idsToDelete
     */
    onDelete(e: Event) {

        const idsToDelete: Set<string> = this.state.idsToDelete;
        const value = (e.target as HTMLInputElement).value;

        if (idsToDelete.has(value)) {

            idsToDelete.delete(value);
        } else {

            idsToDelete.add(value);
        }

        this.setState({
            idsToDelete
        });
    }

    /**
     * Sends idsToDelete to server to be deleted
     */
    onSubmit(target: HTMLFormElement) {

        this.props.deleteArticle({
            variables: {
                ids: [...this.state.idsToDelete],
                password: (target.querySelector('[name=password]') as HTMLInputElement).value
            }
        });
    }

    render() {

        if (!this.props.articles || this.props.articles.length < 1) {
            return null;
        }

        return (
            <UserArticleTable
              articles={this.props.articles}
              onDelete={this.onDelete}
              onSubmit={this.onSubmit}
            />
        );
    }
}


const UserArticleTableContainerWithData = compose(
    graphql(ArticleDelete, {name: 'deleteArticle'})
)(UserArticleTableContainer as any);

export default withApollo(UserArticleTableContainerWithData);