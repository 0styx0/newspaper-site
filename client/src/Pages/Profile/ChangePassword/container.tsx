import * as React from 'react';
import ChangePassword from './';
import { UserUpdate } from '../../../graphql/user';
import { graphql, compose, withApollo } from 'react-apollo';

interface Props {
    updatePassword: Function;
}

export class ChangePasswordContainer extends React.Component<Props, {}> {

    constructor() {
        super();

        this.onSubmit = this.onSubmit.bind(this);
    }

    /**
     * Send data to server
     */
    onSubmit(e: Event) {

        const target = e.target as HTMLFormElement;

        const oldPassword = target.querySelector('[name=password]') as HTMLInputElement;
        const newPassword = target.querySelector('[name=newPassword]') as HTMLInputElement;
        const newPasswordConfirmation = target.querySelector('[name=newPasswordConfirmation]') as HTMLInputElement;

        if (newPassword.value !== newPasswordConfirmation.value ||
          !(oldPassword.value && newPassword.value && newPasswordConfirmation.value)) {

            return;
        }

        this.props.updatePassword({
            variables: {
                password: oldPassword.value,
                newPassword: newPassword.value
            }
        });
    }

    render() {
        return <ChangePassword onSubmit={this.onSubmit} />;
    }
}

const ChangePasswordContainerWithData = compose(
    graphql(UserUpdate, {name: 'updatePassword'}) as any
)(ChangePasswordContainer);

export default withApollo(ChangePasswordContainerWithData);