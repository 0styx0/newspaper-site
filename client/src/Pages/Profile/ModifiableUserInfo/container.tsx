import * as React from 'react';
import { PrivateUserQuery, UserUpdate } from '../../../graphql/user';
import { UserDelete } from '../../../graphql/users';
import { graphql, withApollo, compose } from 'react-apollo';
import { ModifiableUserInfo } from '../shared.interfaces';
import ModifiableUserInfoComponent from './';

interface Props {
    updateUser: Function;
    deleteUser: Function;
    privateUserData: {
        users: [ModifiableUserInfo] // always length = 1
    };
}

interface State {
    updates: {
        twoFactor?: boolean;
        notificationStatus?: boolean;
    };
    delete?: boolean;
}

/**
 * Container, has event listeners and passes data to @see ./index.tsx
 */
export class ModifiableUserInfoContainer extends React.Component<Props, State> {

    constructor() {
        super();

        this.onDelete = this.onDelete.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.onChange = this.onChange.bind(this);

        this.state = {
            updates: {}
        };
    }

    onDelete(e: Event) {

        this.setState({
            delete: (e.target as HTMLInputElement).checked
        });
    }

    /**
     * Saves changes to state.updates
     *
     * @uses `e.target.name`, `e.target.value`
     */
    onChange(e: Event) {

        const target = e.target as HTMLInputElement;

        const updates = Object.assign({}, this.state.updates);

        updates[target.name] = target.checked;

        this.setState({
            updates
        });
    }

    /**
     * Sends `state.updates` to server
     */
    onSubmit(target: HTMLFormElement) {

        if (this.state.delete) {
            return this.deleteUser();
        }

        if (Object.keys(this.state.updates).length > 0) {
            this.props.updateUser({
                variables: Object.assign(this.state.updates, {
                    password: (target.querySelector('[name=password]') as HTMLInputElement).value
                })
            });
        }
    }

    /**
     * Sends id of user to server so user can be deleted
     */
    deleteUser() {

        this.props.deleteUser({
            variables: {
                ids: [this.props.privateUserData.users[0].id]
            }
        });
    }

    render() {

        if (!this.props.privateUserData || !this.props.privateUserData.users) {
           return null;
        }

        return (
            <ModifiableUserInfoComponent
              onSubmit={this.onSubmit}
              onChange={this.onChange}
              onDelete={this.onDelete}
              {...this.props.privateUserData.users[0]}
            />
        );
    }
}

const ModifiableUserInfoContainerWithData = compose(
    graphql(PrivateUserQuery, {
        name: 'privateUserData',
        options: {
            variables: {
                profileLink: 'meiselesd2018'
            }
        }
    }),
    graphql(UserUpdate, {name: 'updateUser'}),
    graphql(UserDelete, {name: 'deleteUser'})
)(ModifiableUserInfoContainer as any);

export default withApollo(ModifiableUserInfoContainerWithData);
