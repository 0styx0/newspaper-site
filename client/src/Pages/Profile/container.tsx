import * as React from 'react';
import { UserQuery } from '../../graphql/user';
import { graphql, compose, withApollo } from 'react-apollo';
import Profile from './';
import { Article, PublicUserInfo } from './shared.interfaces';

interface Props {
    data: {
        users: {
            articles: Article[];
        } & PublicUserInfo
    };
    client: {
        query: ( params: { query: typeof UserQuery, variables: { profileLink: string; } } ) => Promise<Props>;
    };
}

interface State {
    articles: Article[];
    user: PublicUserInfo;
}

class ProfileContainer extends React.Component<Props, State> {

    public state: State;

    constructor(props: Props) {
        super(props);

        this.state = {
            user: {} as PublicUserInfo,
            articles: [] as Article[]
        };

    }

    async componentWillMount() {

        const path = window.location.pathname.split('/');

        const { data } = await this.props.client.query({
            query: UserQuery,
            variables: {
                profileLink: path[path.length - 1]
            }
        });

        this.setState({
            user: data.users[0],
            articles: data.users[0].articles
        });
    }

    render() {

        if (!this.state.user) {
            return null;
        }

        const articles = this.state.articles as Article[];
        const user = this.state.user as PublicUserInfo;

        return (
            <Profile
                articles={articles}
                user={user}
            />
        );
    }
}

const ProfileContainerWithData = compose(
    graphql(UserQuery, {
        options: {
            variables: {
                profileLink: window.location.pathname.split('/')[window.location.pathname.split('/').length - 1]
            }
        }
    })
)(ProfileContainer);

export default withApollo(ProfileContainerWithData as any) as any;
