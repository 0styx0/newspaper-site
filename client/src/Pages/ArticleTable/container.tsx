import * as React from 'react';
import { ArticleQuery, ArticleUpdate, ArticleDelete } from '../../graphql/articles';
import { compose, graphql, withApollo } from 'react-apollo';

import ArticleTable from './';

export interface Article {
    tags: {
            all: string[];
        };
    url: string;
    id: string;
    displayOrder: number;
    dateCreated: string;
    views: number;
    author: {
        fullName: string;
        profileLink: string;
    };
}

export interface Issue {
    num: number;
    max: number;
}

interface Props {
    data: {
        loading: boolean;
        issues?: [
            Issue & {
                articles: Article[]
            }
        ]; // will never be more length than 1
    };
    client: {
        query: ( params: { query: typeof ArticleQuery, variables: { issue: number | null; } } ) => Promise<Props>;
    };
    updateArticle: Function;
    deleteArticle: Function;
}

interface State {
    issue: Issue;
    articles: Article[];
    updates: {
        idsToDelete: Set<string>;
        displayOrder: Map<string, number>; // id => info
        tags: Map<string, string[]>;
    };
}

/**
 * Controller of ArticleTable (@see ./index)
 *
 * Gets data from server, then renders ArticleTable with it.
 * This holds listeners for all events that @see ArticleTable can fire
 */
export class ArticleTableContainer extends React.Component<Props, State> {

    constructor() {
        super();

        this.putData = this.putData.bind(this);
        this.convertPropsToState = this.convertPropsToState.bind(this);
        this.onChange = this.onChange.bind(this);
        this.onDelete = this.onDelete.bind(this);
        this.onSubmit = this.onSubmit.bind(this);



        this.state = {
            issue: {} as Issue,
            articles: [],
            updates: {
                idsToDelete: new Set<string>(),
                displayOrder: new Map<string, number>(),
                tags: new Map<string, string[]>()
            }
        };
    }

    /**
     * @param num - articles from what issue to get. If null, get latest
     *
     * Gets data needed for table (@see State.Article interface) and changes page url to modifyArticles/currentIssue
     *
     * Depends on window.location.pathname
     */
    async putData(num: number | null = null) {

        const lastPath = +window.location.pathname.split('/').slice(-1)[0];

        num = (isNaN(lastPath) || ((!isNaN(+num!)) && num)) ? num : lastPath;

        this.props.client.query({
            query: ArticleQuery,
            variables: {
                issue: num
            }
        }).then((data: Props) => {
            this.convertPropsToState(data);
        });

        window.history.pushState(
            {},
            `Issue ${num}`, isNaN(lastPath) ?
              `modifyArticles/${num}` :
             num + ''
        );
    }

    /**
     * Seeds state with graphql props
     */
    componentWillReceiveProps(newProps: Props) {
        this.convertPropsToState(newProps);
    }

    /**
     * Sets `state.issue` and `state.articles` to values extracted from props
     */
    convertPropsToState(props: Props) {

        if (!props.data.issues) {
            return;
        }

        let {max, num} = props.data.issues![0];

        this.setState({
            issue: {
                max,
                num
            },
            articles: props.data.issues![0].articles
        });
    }

    /**
     * Saves changes to state so can be submitted later on. This is for updates only. @see onDelete for deletions
     *
     * @uses `e.target.name` as index of state.updates
     * @uses `e.target.value` as the value
     */
    onChange(e: Event, article: Article) {

        e.stopPropagation();
        e.preventDefault();

        const target = e.target as HTMLInputElement | HTMLSelectElement;

        const stateUpdate = this.state.updates;

        const modifyMapCopy = new Map(stateUpdate[target.name]);
        let value: number | string[];

        if ((target as HTMLSelectElement).selectedOptions) { // for tags

            value = Array.from((target as HTMLSelectElement).selectedOptions)
                    .map(option => option.value)
                    .slice(-3)
                    .sort(); // using sort to make it easier to test, no actual reason besides that
        } else {
            value = +target.value;
        }

        modifyMapCopy.set(article.id, value);

        stateUpdate[target.name] = modifyMapCopy;

        this.setState({
            updates: stateUpdate
        });
    }

    /**
     * Adds/removes user's id from list of users to delete
     *
     * @uses e.target.checked as way to see if should add or remove e.target.value from list
     *
     * @param e {HTMLCheckboxEvent}
     */
    onDelete(e: Event) {

        const stateUpdate = this.state.updates;

        const target = e.target as HTMLInputElement;

        target.checked ? stateUpdate.idsToDelete.add(target.value) : stateUpdate.idsToDelete.delete(target.value);

        this.setState({
            updates: stateUpdate
        });
    }

    /**
     * Sends data from `state.updates` to server
     */
    onSubmit(e: Event) {

        e.stopPropagation();
        e.preventDefault();

        // separating update and delete into functions just to show they're separate stuff
        const submitUpdated = () => {

            const { displayOrder, tags } = this.state.updates;

            if (displayOrder.size + tags.size < 1) {
                return;
            }

            let data: {id: string, displayOrder?: number, tags?: string[]}[] = [...displayOrder].map(mapping => ({

                id: mapping[0],
                displayOrder: mapping[1]
            }));

            [...tags].forEach(mapping => {

                const idIdx = data.findIndex(elt => elt.id === mapping[0]);

                if (idIdx !== -1) {
                    data[idIdx].tags = mapping[1];
                } else {
                    data.push({
                        id: mapping[0],
                        tags: mapping[1]
                    });
                }
            });

            this.props.updateArticle({
                variables: {
                    data
                }
            });
        };

        const submitDeleted = () => {

            if (this.state.updates.idsToDelete.size > 0) {

                this.props.deleteArticle({
                    variables: {
                        ids: [...this.state.updates.idsToDelete]
                    }
                });
            }
        };

        submitUpdated();
        submitDeleted();
    }

    render() {

        if (this.state.articles.length < 1) {
            return null;
        }

        return (
            <ArticleTable
                articles={this.state.articles}
                key={this.state.issue.num}
                issue={this.state.issue}
                onChange={this.onChange}
                onDelete={this.onDelete}
                onUpdate={(e: Event) => this.putData(+(e.target as HTMLInputElement).value)}
                onSubmit={this.onSubmit}
            />
        );
    }
}

const ArticleTableContainerWithData = compose(
    graphql(ArticleQuery, {
        options: {
            variables: {
                issue: +window.location.pathname.split('/').slice(-1)[0]
            }
        }
    }),
    graphql(ArticleUpdate, {name: 'updateArticle'}),
    graphql(ArticleDelete, {name: 'deleteArticle'})
)(ArticleTableContainer as any);

export default withApollo(ArticleTableContainerWithData);