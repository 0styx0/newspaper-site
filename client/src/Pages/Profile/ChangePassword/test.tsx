import * as React from 'react';
import { ChangePasswordContainer, Props } from './container';
import * as renderer from 'react-test-renderer';
import * as casual from 'casual';
import * as sinon from 'sinon';
import { mount, ReactWrapper } from 'enzyme';
import { submitForm } from '../../../tests/enzyme.helpers';

function setup(mockGraphql: {updatePassword?: Function} = {}) {

    return mount(
        <ChangePasswordContainer
            updatePassword={mockGraphql.updatePassword ? mockGraphql.updatePassword : () => true}
        />
    );
}

describe('<ChangePasswordContainer>', () => {

    describe('snapshots', () => {

         it ('should render correctly', () => {

            const tree = renderer.create(

                <ChangePasswordContainer updatePassword={() => true} />
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('submitting password', () => {

        let wrapper: ReactWrapper<Props, {}>;

        /**
         * Set value of inputs and submits the form
         */
        function setValues(oldPassword: string, newPassword: string, newPasswordConfirmation: string) {

            (wrapper.find('input[name="password"]').instance() as {} as HTMLInputElement)
                .value = oldPassword;
            (wrapper.find('input[name="newPassword"]').instance() as {} as HTMLInputElement)
                .value = newPassword;
            (wrapper.find('input[name="newPasswordConfirmation"]').instance() as {} as HTMLInputElement)
                .value = newPasswordConfirmation;

            submitForm(wrapper);
        }

        it('submits all data needed if valid', () => {

            const newPassword = casual.string;
            const oldPassword = casual.string;

            wrapper = setup({updatePassword: (data: {variables: { newPassword: string, password: string}}) => {

                expect(data.variables.newPassword).toBe(newPassword);
                expect(data.variables.password).toBe(oldPassword);
            }});

            setValues(oldPassword, newPassword, newPassword);
        });

        it('is not submitted if passwords do not match', () => {

            const spy = sinon.spy();

            wrapper = setup({updatePassword: spy});

            setValues(casual.string, casual.string, casual.string);

            expect(spy.called).toBeFalsy();
        });

        it('is not submitted if an input is empty', () => {

            const newPassword = casual.string;
            const oldPassword = casual.string;

            const spy = sinon.spy();

            wrapper = setup({updatePassword: spy});

            const values = [oldPassword, newPassword, newPassword];

            values[casual.integer(0, 2)] = '';

            setValues(values[0], values[1], values[2]);

            expect(spy.called).toBeFalsy();
        });
    });
});
