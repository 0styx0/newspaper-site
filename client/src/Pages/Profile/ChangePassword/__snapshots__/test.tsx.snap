// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePasswordContainer> snapshots should render correctly 1`] = `
<section
  className="container "
>
  <h1>
    Change Password
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      New Password: 
      <input
        name="newPasswordConfirmation"
        onBlur={[Function]}
        pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\\\\d)(?=.*).{6,}$"
        required={true}
        type="password"
      />
      <span
        className="danger"
      >
        *
      </span>
    </label>
    <label>
      Confirm Password: 
      <input
        name="newPassword"
        onBlur={[Function]}
        pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\\\\d)(?=.*).{6,}$"
        required={true}
        type="password"
      />
      <span
        className="danger"
      >
        *
      </span>
    </label>
    <label>
      Old Password: 
      <input
        name="password"
        onBlur={[Function]}
        pattern="^(?=.*[a-z])(?=.*[A-Z])(?=.*\\\\d)(?=.*).{6,}$"
        required={true}
        type="password"
      />
      <span
        className="danger"
      >
        *
      </span>
    </label>
    <input
      type="submit"
      value="Change Password"
    />
  </form>
</section>
`;
